### PR TITLE
Add a workaround for GH checkout action

### LINKED ISSUE #302: CI: git describe is not getting the tag being created in the create event
In #301 we migrated to GitHub Actions, and we are creating and archiving the Debian package as an artifact, but `git` is not getting the latest tag when doing a `describe`. I'm not sure if the `checkout` step is not fetching the latest tag for some reason or what else is going on.

Re-running the job doesn't fix anything so it doesn't seem to be a timing issue and it's very weird because it does find the previous tag, so it is also fetching tags.

### PR DESCRIPTION
**Pull request by llucax**
The action is not properly fetching annotated tags, so git describe
fails.

Fixes #302.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (jwilk) [inline .github/workflows/ci.yml]:
```suggestion
        run: /usr/bin/git -c protocol.version=2 fetch --prune --progress 
```

Comment (llucax) [reply .github/workflows/ci.yml]: Good catch, thanks!

**Review by ibuclaw — DISMISSED** [review]

**Comment by llucax** [comment]
Rebased, need a new approval @ibuclaw. I can remove this restriction (automatically dismiss approvals if there are new pushes) if it's too annoying (I think in the general case is useful to make sure no weird stuff gets in without a review but for simple rebases on top of the current v2.x.x like this it is a bit annoying).

File: .github/workflows/ci.yml
Comment (ibuclaw) [inline .github/workflows/ci.yml:15]: I assume that there's no need for `run: |` or `\` before end of line for the multiline yaml (I forget the actual syntax, so I might be wrong).

**Review by ibuclaw — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (llucax) [reply .github/workflows/ci.yml:15]: Correct. I also tested it on my fork with a tag and it works.

**Pull request merged**
